### PR TITLE
Whisper models

### PR DESCRIPTION
**Pull request by mahenning**
Added the whisperx model to the Transcriber class.

Made the choice that `Transcriber` is not an ABC class, because now one can use `Transcriber.load_model(..., whisper_type, ...)` to automatically load the right whisper/whisperx model.  
Tested with ScrAIbe-WebUI docker container and a transcript of a few mp3/mp4.

To load the appropiate model type, add e.g.
> models:
>    whisper_type: whisperx
to the config.yaml. Default is whisper.

### CODE REVIEW

**Comment by mahenning** [comment]
Last two commits added a large part of formatting fixes and nothing much else, maybe it's easier to check by commit.

**Review by JSchmie — CHANGES_REQUESTED** [review]
Overall good, but there are two major issues. 
The first one, please ensure code coverage, you made a major change without testing it properly.
The second is that you run the classes in a cycle, which is confusing and causes problems.
For that reason, I would suggest making an extra function to get the Whisper Classes.

**Comment by mahenning** [comment]
> The second is that you run the classes in a cycle, which is confusing and causes problems.

What do you mean by that? What cycle?

**Comment by JSchmie** [comment]
You use `Transcriber`  as a base class. But later on, you use `Transcriber` to call the inherit classes using a Class method. A class method is designed to init a Class by a function, not to init another class. 
In principle, I am fine with using `Transcriber` to load other classes again but then use a `@staticmethod` instead.

File: scraibe/transcriber.py [hunk 1]
```diff
@@ -143,10 +141,103 @@ def load_model(cls,
...
+            in_memory (bool, optional): Whether to load model in memory. 
+                                        Defaults to False.
+            args: Additional arguments only to avoid errors.
+            kwargs: Additional keyword arguments only to avoid errors.
+
+        Returns:
+            Transcriber: A Transcriber object initialized with the specified model.
+        """
+        if whisper_type.lower() == 'whisper':
+            _model = WhisperTranscriber.load_model(
```
[the remark this answers is not in the document]
Comment (mahenning) [reply scraibe/transcriber.py:159]: In your implementation of Transcriber, you used a class method for `load_model`. Why not here? Should WhisperTranscriber/WhisperxTranscriber `load_model` be called directly? How would this confuse users? A base class call can be used for scalability if we would want ot add even more whisper alternatives. (Maybe not the best idea to add half a dozen whisper implementations as deps)

(Not to nitpick too much, but in general a "should do this" is always more helpful to me than a "should not do this". I still have no idea what exactly I should do here.)

File: scraibe/autotranscript.py [hunk 2]
```diff
@@ -81,12 +85,13 @@ def __init__(self,
...
         """
-        
-        
+
         if whisper_model is None:
-            self.transcriber = Transcriber.load_model("medium", **kwargs)    
+            self.transcriber = Transcriber.load_model(
+                "medium", whisper_type, **kwargs)
         elif isinstance(whisper_model, str):
-            self.transcriber = Transcriber.load_model(whisper_model, **kwargs)
```
[the remark this answers is not in the document]
Comment (mahenning) [reply scraibe/autotranscript.py:89]: Yes. I added the return type hint to adress this. Type hint is now
```
-> Union[WhisperTranscriber, WhisperXTranscriber]```

File: scraibe/transcriber.py [hunk 3]
```diff
@@ -171,17 +262,116 @@ def _get_whisper_kwargs(**kwargs) -> dict:
...
+
         if (language := kwargs.get("language")):
-            whisper_kwargs["language"] = language 
-        
+            whisper_kwargs["language"] = language
+
+        return whisper_kwargs
+
+
+class WhisperXTranscriber(Transcriber):
```
[the remark this answers is not in the document]
Comment (mahenning) [reply scraibe/transcriber.py:285]: Done.

File: scraibe/transcriber.py [hunk 4]
```diff
@@ -143,10 +141,103 @@ def load_model(cls,
...
+        Returns:
+            dict: Keyword arguments for whisper model.
+        """
+        pass
+
+    def __repr__(self) -> str:
+        return f"Transcriber(model_name={self.model_name}, model={self.model})"
+
+
+class WhisperTranscriber(Transcriber):
```
[the remark this answers is not in the document]
Comment (mahenning) [reply scraibe/transcriber.py:185]: Done.

**Comment by mahenning** [comment]
Changed decorator of `load_model` from `@classmethod` to `@staticmethod`.

File: test/test_transcriber.py
[the remark this answers is not in the document]
Comment (mahenning) [reply test/test_transcriber.py:1]: Is this right now what you meant? Extended the tests, and test for each subclass. Also added test for wrong input (could be also extended for more wrong input)

**Pull request merged**